Previously missing bg95 shutdown script added to modem_cellular.c

acf879 opened the pull request:
Prior to this addition the bg95 cellular modem did not have a shutdown script and instead had a NULL in place of it. According to the hardware design document and AT command guide from quectel a command is available to turn off the modem. This was added as the shutdown script to allow the modem to restart without issues.

rerickson1 commented:
You just need to rebase your change onto main so this PR only contains your one commit and not the merge commit.

jukkar commented on drivers/modem/modem_cellular.c:
```diff
@@ -1854,7 +1854,7 @@ MODEM_CHAT_SCRIPT_CMDS_DEFINE(quectel_bg95_init_chat_script_cmds,
 			      MODEM_CHAT_SCRIPT_CMD_RESP_NONE("AT+CMUX=0,0,5,127", 300));
```
Please fix the commit subject to something like `modem: bg95: Added cellular modem shutdown script`
Also check the compliance check warnings.

acf879 replied:
I changed the name to that now and I think I fixed the compliance check warnings for the modem_cellular.c file. I will check back once it is finished with its tests and fix anything else it flags.

acf879 committed:
modem: bg95: Added cellular modem shutdown script

The BG95 cellular modem was previously missing a
  shutdown script and had a NULL placed where it
  should be. According to the quectel AT commands
  document and hardware design document `AT+POWD`
  can be found. When `AT+POWD=1` shuts down
  immediately. When `AT+POWD=0` the modem shuts
  down through the normal shutdown procedure.
  This allows the use of `pm suspend` through the
  shell command in the example
  `zephyr/samples/net/cellular_modem` to turn off
  the modem as expected. This also allows the modem
  init script to be able to be called following the
  power down of the modem.

Signed-off-by: Ashton Fritz <ashton.fritz@usask.ca>
